File: src/main/java/cn/hselfweb/mes/arithmetic/entity/XLList.java
package cn.hselfweb.mes.arithmetic.entity;

import cn.hselfweb.mes.enity.CraftExtend;
import com.google.gson.Gson;
import com.google.gson.stream.JsonReader;

import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.List;

public class XLList {
    private static final String dataPath = "data.json";
    private int DeviceNumber;
    private List<List<CraftExtend>> list_sum = new ArrayList<List<CraftExtend>>();

    public int getDeviceNumber() {
        return DeviceNumber;
    }

    public void setDeviceNumber(int deviceNumber) {
        DeviceNumber = deviceNumber;
    }

    public List<List<CraftExtend>> getList_sum() {
        return list_sum;
    }

    public void setList_sum(List<List<CraftExtend>> list_sum) {
        this.list_sum = list_sum;
    }

    /**
     * 实例化  调度工序列表
     *
     * @throws FileNotFoundException
     */
    public static final XLList Load() throws FileNotFoundException {
        Gson gson = new Gson();
        JsonReader reader = new JsonReader(new InputStreamReader(new FileInputStream(dataPath)));
        XLList ListC = gson.fromJson(reader, XLList.class);

        int gxCount = ListC.getList_sum().size();
        int gjCount = ListC.getList_sum().get(0).size();
        for (int i = 0; i < gjCount; i++) {
            for (int j = 0; j < gxCount; j++) {
                ListC.getList_sum().get(j).get(i).setGid(j + 1);
                if (j != 0) {
                    //设置前序工序
                    ListC.getList_sum().get(j).get(i).setExtendCraftByPreEcId(ListC.getList_sum().get(j - 1).get(i));
                }
                if (j < gxCount - 1) {
                    //设置后序工序
                    ListC.getList_sum().get(j).get(i).setExtendCraftByAftEcId(ListC.getList_sum().get(j + 1).get(i));
                }

            }
        }
        return ListC;
    }

}

File: src/main/java/cn/hselfweb/mes/arithmetic/entity/XLList.kt
package cn.hselfweb.mes.arithmetic.entity

import cn.hselfweb.mes.enity.CraftExtend
import com.google.gson.Gson
import com.google.gson.stream.JsonReader

import java.io.FileInputStream
import java.io.FileNotFoundException
import java.io.InputStreamReader
import java.util.ArrayList

class XLList {
    var deviceNumber: Int = 0
    var list_sum: MutableList<MutableList<CraftExtend>> = ArrayList()

    companion object {
        private val dataPath = "data.json"
        /**
         * 实例化  调度工序列表
         * @throws FileNotFoundException
         */
        @Throws(FileNotFoundException::class)
        fun Load(): XLList {
            val gson = Gson()
            val reader = JsonReader(InputStreamReader(FileInputStream(dataPath)))
            val ListC = gson.fromJson<XLList>(reader, XLList::class.java)

            val gxCount = ListC.list_sum.size
            val gjCount = ListC.list_sum[0].size
            for (i in 0 until gjCount) {
                for (j in 0 until gxCount) {
                    ListC.list_sum[j][i].gid = j + 1
                    if (j != 0) {
                        //设置前序工序
                        ListC.list_sum[j][i].extendCraftByPreEcId = ListC.list_sum[j - 1][i]
                    }
                    if (j < gxCount - 1) {
                        //设置后序工序
                        ListC.list_sum[j][i].extendCraftByAftEcId = ListC.list_sum[j + 1][i]
                    }

                }
            }
            return ListC
        }
    }

}
